### PR TITLE
refactor(container): expose an api for confirm dialog show/hide

### PR DESCRIPTION
**Pull request by jlame**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
The host app was the one responsible for changing the state to modify the visibility/conf of the confirm dialog.

**What is the new behavior?**
Now the container exposes two reducing function which returns the new state to the host app's reducer.

**Does this PR introduce a breaking change?**

- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...
The impact for the moment is on semantic-ee-frontend as it is the only one who uses the confirm dialog container for the moment.
A PR is already awaiting for this PR to be merged. In that PR, we just call the exposed show/hide function instead of doing the treatment in the reducer itself. 
So right after merging this PR, we'll have to merge other on semantic-ee-frontend.

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

File: packages/containers/src/ConfirmDialog/showHideConfirmDialog.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export function hideConfirmDialog(state) {
+	//hiding the modal
+	const path = ['ConfirmDialog', 'ConfirmDialog', 'show'];
+	const newState = { ...state };
+	newState.cmf.components = state.cmf.components.setIn(path, false);
+	return newState;
+}
+
```
Comment (jsomsanith) [inline packages/containers/src/ConfirmDialog/showHideConfirmDialog.js]: Your 2 apis are not consistent.
`showConfirmDialog` should show the dialog and `hideConfirmDialog` should hide it.

They should use a common code (from `src/state.js` ? (see my global review comment)) and only customise the value (true/false).

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://221.talend.surge.sh)

**Pull request merged**